spring-integration-core/src/main/java/org/springframework/integration/json/JsonPropertyAccessor.java

```diff
@@ -50,16 +56,32 @@
 	 */
 	private static final Class<?>[] SUPPORTED_CLASSES =
 			new Class<?>[] {
-					String.class,
-					ToStringFriendlyJsonNode.class,
-					ArrayNodeAsList.class,
-					ObjectNode.class,
-					ArrayNode.class
-			};
+				String.class,
+				JsonNodeWrapper.class,
+				JsonNode.class
+		};
 
 	// Note: ObjectMapper is thread-safe
 	private ObjectMapper objectMapper = new ObjectMapper();
 
+	public JsonPropertyAccessor() {
+		this((ConverterRegistry) DefaultConversionService.getSharedInstance());
+	}
+
+	public JsonPropertyAccessor(ConverterRegistry converterRegistry) {
+		converterRegistry.addConverter(new GenericConverter() {
+			@Override
+			public Set<ConvertiblePair> getConvertibleTypes() {
+				return Collections.singleton(new ConvertiblePair(JsonNodeWrapper.class, JsonNode.class));
```

Comment:
Why do we need this converter?

If that, shouldn't we introduce it alongside with that `ToStringFriendlyJsonNodeToStringConverter`?
On the other hand: do we still need that `ToStringFriendlyJsonNodeToStringConverter` with this whole your change?